app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  # Columns
+  validates :uid, presence: true, uniqueness: true
+  validates :username, presence: true
+  validates :email, presence: true, uniqueness: true
+  validates :verified, inclusion: { in: [true, false] }
+  validates :login_count, numericality: { only_integer: true }
+
+  # Relations
+  has_many :authentication_events, foreign_key: 'user_id'
+
+  # Class methods
+  def self.email_exists?(email)
+    exists?(email: email)
+  end
```

Comment:
The `email_exists?` method should be indexed to improve query performance, especially if the `users` table grows large.